core/templates/components/skill-selector/select-skill-modal.template.html

```diff
@@ -10,6 +10,6 @@ <h3>
   </select-skill>
 </div>
 <div class="modal-footer">
-  <button class="btn btn-success protractor-test-confirm-skill-selection-button" ng-click="confirm(selectedSkillId)" ng-disabled="!selectedSkillId">Done</button>
+  <button class="btn btn-success protractor-test-confirm-skill-selection-button" ng-click="confirm(selectedSkillId, preModalCloseHook)" ng-disabled="!selectedSkillId">Done</button>
```

Comment:
Why don't you call `save` instead of `confirm` here and the in the `save` method you call `confirm(selectedSkillId)` after the `for` loop. Then these changes won't be necessary (I think we should avoid adding more features than just closing and canceling the modal).

```suggestion
  <button class="btn btn-success protractor-test-confirm-skill-selection-button" ng-click="save()" ng-disabled="!selectedSkillId">Done</button>
```

Reply:
@marianazangrossi, The problem with this approach is that It works only for the question editor.(the `save()` function is implemented only in question-editor).
The select-skill modal is used in other places too.
I was thinking of a more general solution like what @seanlip suggested.

Reply:
On looking at the code, I actually agree with @marianazangrossi. The modal controllers are available and can contain the additional customization needed, so the individual controllers should handle it, not the general infrastructure. Would that work @donosco98?

Reply:
Also @donosco98 next time please update the assignees to the PR if you want someone to reply. Thanks!

Reply:
done